Lesson4/step2/templates/menu.html

```diff
@@ -1,114 +1,133 @@
-							<p class="menu-price">{{i.price}}</p>
-							<a href='{{url_for('editMenuItem', restaurant_id = restaurant.id, menu_id=i.id ) }}'>Edit</a> | 
-							<a href='{{url_for('deleteMenuItem', restaurant_id = restaurant.id, menu_id=i.id ) }}'>Delete</a>
-						</div>
-						{% endif %}
-					{% endfor %}
-				<h2>Beverages</h2>
-					{% for i in items %}
-						{% if i.course == 'Beverage' %}
-						<div class="menu-item">
-							<h3>{{i.name}}</h3>
-							<p>{{i.description}}</p>
-							<p class="menu-price">{{i.price}}</p>
-							<a href='{{url_for('editMenuItem', restaurant_id = restaurant.id, menu_id=i.id ) }}'>Edit</a> | 
-							<a href='{{url_for('deleteMenuItem', restaurant_id = restaurant.id, menu_id=i.id ) }}'>Delete</a>
-						</div>
-						{% endif %}
-					{% endfor %}
-			</div>
-			<div class="col-md-1"></div>
-		</div>
-	{% endif %}
+                <ul>
+                    {% for message in messages %}
+                        <li><strong> {{ message }} </strong></li>
+                    {% endfor %}
+                </ul>
+            {% endif %}
+        {% endwith %}
+
+    </div>
+    {% if login_session.user_id == restaurant.user_id %}
+        <div class="row padding-top padding-bottom">
+            <div class="col-md-1"></div>
+            <div class="col-md-11 padding-none">
+
+                <a href="{{ url_for('edit_restaurant', restaurant_id = restaurant.id ) }}">
+                    <button class="btn btn-default" id="new-menu-item">
+                        <span class="glyphicon glyphicon-pencil" aria-hidden="true"></span>Edit Restaurant
+                    </button>
+                </a>
+                <a href="{{ url_for('new_menu_item', restaurant_id = restaurant.id ) }}">
+                    <button class="btn btn-default" id="new-menu-item">
+                        <span class="glyphicon glyphicon-glass" aria-hidden="true"></span>Add Menu Item
+                    </button>
+                </a>
+                <a href="{{ url_for('delete_restaurant', restaurant_id = restaurant.id ) }}">
+                    <button class="btn btn-default delete" id="delete-restaurant">
+                        <span class="glyphicon glyphicon-trash" aria-hidden="true"></span>Delete Restaurant
+                    </button>
+                </a>
+            </div>
+            <div class="col-md-7"></div>
+        </div>
+    {% endif %}
+
+    {% if items !=[] %}
+        <div class="row">
+            <div class="col-md-1"></div>
+            <div class="col-md-3">
+                <h2>Appetizers</h2>
+                {% for i in items %}
+                    {% if i.course == 'Appetizer' %}
+                        <div class="menu-item">
+                            <h3>{{ i.name }}</h3>
+                            <p>{{ i.description }}</p>
+                            <p class="menu-price">{{ i.price }}</p>
+                            {% if restaurant.user_id == login_session.user_id %}
```

Comment:
By capturing the login_session in the controller, we can make sure that users don't have to duplicate their HTML pages.